ambry-store/src/main/java/com.github.ambry.store/PersistentIndex.java

```diff
@@ -219,14 +222,35 @@ public boolean accept(File dir, String name) {
   }
 
   /**
+   * Loads all index segment files that refer to segments in the log.
+   * @return all index segment files that refer to segments in the log.
+   * @throws StoreException if index segment files could not be listed from the directory.
+   */
+  private List<File> getIndexFiles() throws StoreException {
+    List<File> indexFiles = new ArrayList<>();
+    LogSegment logSegment = log.getFirstSegment();
```

Comment:
does it make sense to have an iterator for the log segments?
Also, I don't have the whole context on when this method will be called, but how do you - or do you at all need to - ensure that you get a point-in-time view of the list of log segments while iterating?